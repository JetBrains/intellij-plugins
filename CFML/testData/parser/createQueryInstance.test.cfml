<cfscript>
  var q =  new Query(sql = sql, datasource = dsn);
</cfscript>