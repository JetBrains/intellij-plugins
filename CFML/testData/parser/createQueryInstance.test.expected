CfmlFile:createQueryInstance.cfml
  Tag
    PsiElement(OPENER)('<')
    PsiElement(CF_TAG_NAME)('cfscript')
    PsiElement(R_ANGLEBRACKET)('>')
    Assignment
      PsiElement(VAR_KEYWORD)('var')
      ReferenceExpression
        PsiElement(IDENTIFIER)('q')
      PsiElement(ASSIGN)('=')
      NewExpression
        PsiElement(IDENTIFIER)('new')
        COMPONENT_CONSTRUCTOR_CALL
          COMPONENT_REFERENCE
            PsiElement(IDENTIFIER)('Query')
          ArgumentList
            PsiElement(L_BRACKET)('(')
            Assignment
              Argument sql
                PsiElement(IDENTIFIER)('sql')
              PsiElement(ASSIGN)('=')
              ReferenceExpression
                PsiElement(IDENTIFIER)('sql')
            PsiElement(COMMA)(',')
            Assignment
              Argument datasource
                PsiElement(IDENTIFIER)('datasource')
              PsiElement(ASSIGN)('=')
              ReferenceExpression
                PsiElement(IDENTIFIER)('dsn')
            PsiElement(R_BRACKET)(')')
    PsiElement(SEMICOLON)(';')
    PsiElement(LSLASH_ANGLEBRACKET)('</')
    PsiElement(CF_TAG_NAME)('cfscript')
    PsiElement(CLOSER)('>')